pkg/slo-controller/nodeslo/extender_plugin.go

```diff
@@ -0,0 +1,102 @@
+/*
+Copyright 2022 The Koordinator Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nodeslo
+
+import (
+	"fmt"
+	"reflect"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/klog/v2"
+
+	"github.com/koordinator-sh/koordinator/apis/extension"
+	slov1alpha1 "github.com/koordinator-sh/koordinator/apis/slo/v1alpha1"
+)
+
+var (
+	globalNodeSLOMergedExtender = map[string]NodeSLOMergedPlugin{}
+)
+
+func RegisterNodeSLOMergedExtender(name string, extender NodeSLOMergedPlugin) error {
+	if _, exist := globalNodeSLOMergedExtender[name]; exist {
+		return fmt.Errorf("node merged plugin %s already exist", name)
+	}
+	globalNodeSLOMergedExtender[name] = extender
+	return nil
+}
+
+func calculateExtensionsCfgMerged(oldCfg extension.ExtensionCfg, configMap *corev1.ConfigMap, recorder record.EventRecorder) extension.ExtensionCfg {
+	oldCfgCopy := *oldCfg.DeepCopy()
```

Comment:
need checking oldCfg == nil

Reply:
In order to be consistent with the serires functions "calculatexxxxCfgMerged", oldCfg here is a struct instead of a pointer. It's no need to check if oldCfg is nil.